src/hooks/useAppVersion.js

```diff
@@ -9,7 +11,15 @@ function formatAppVersion(appVersion = VersionNumber.appVersion, update) {
 }
 const defaultAppVersion = formatAppVersion();
 
+let codepushLabel = 'None';
+
+codePush.getUpdateMetadata().then(update => {
+  if (update) {
+    codepushLabel = update.label;
+  }
+});
+
 export default function useAppVersion() {
   const [version] = useState(defaultAppVersion);
-  return version;
+  return [version, codepushLabel];
```

Comment:
It doesn't matter at all but for me it would look clearer if the `getUpdateMetadata()` was called inside the hook in `useEffect()`. Now it's resolved on module import while my intuition about hooks would suggest that resolving the info will take place during hook execution.

But it's just my feeling and I am not sure if anyone shares this opinion.

Reply:
nah, because the answer is async. It takes a few so no chance this will be outdated when calling this function. However, we don't want another rerender if this is not essential. This value will not change in runtime